### PR TITLE
fix http message parser and javascript lib for rdsn calls 

### PR DESCRIPTION
**Pull request by imzhenyu**
fix http message parser due to:
- OPTIONAL is used for http request when cross-domain http msg is used
- custom headers are forbidden when cross-domain http msgs are used
- http msgs are splitted or batched in one TCP read (due to TCP streaming) (cannot assume each read only returns one whole msg)

fix unnecessary "parse failed" due to uninitialized read_next in parsers::get_message_on_receive

### CODE REVIEW

**Comment by imzhenyu** [comment]
This is to fix #420 , now cli works, but other components in web studio are still broken, to be fixed later.

File: src/core/tools/common/http_message_parser.cpp [hunk 1]
```diff
@@ -57,47 +55,60 @@ char(&ArraySizeHelper(T(&array)[N]))[N];
...
-            derror("invalid url '%.*s'", length, at);
+            derror("too long rpc code in url %s", url.c_str());
             return 1;
         }
-        if (cur->length() >= DSN_MAX_TASK_CODE_NAME_LENGTH)
+        strcpy(hdr->rpc_name, args[0].c_str());
+        
+        // hash
+        char *end;
+        hdr->client.hash = std::strtoull(args[1].c_str(), &end, 10);
```
Comment (qinzuoyan) [inline src/core/tools/common/http_message_parser.cpp:103]: do not check the end

Comment (imzhenyu) [inline src/core/tools/common/http_message_parser.cpp:103]: done.

**Comment by qinzuoyan** [comment]
looks good to me

**Pull request merged**